src/Symfony/Component/String/AbstractUnicodeString.php

```diff
@@ -0,0 +1,472 @@
+            }
+        }
+
+        return new static($string);
+    }
+
+    /**
+     * Generic UTF-8 to ASCII transliteration.
+     *
+     * Install the intl extension for best results.
+     *
+     * @param string[] $rules See "*-Latin" rules from Transliterator::listIDs()
+     */
+    public function ascii(array $rules = []): self
+    {
+        $str = clone $this;
+        $s = $str->string;
+        $str->string = '';
+        $step = 0;
+
+        if (\function_exists('transliterator_transliterate')) {
+            $rules[] = '[:nonspacing mark:] remove';
+            $rules[] = 'any-latin';
+        }
+
+        while (\strlen($s) !== $i = strspn($s, self::ASCII)) {
+            if (0 !== $i) {
+                $str->string .= substr($s, 0, $i);
+                $s = substr($s, $i);
+            }
+
+            if (1 === ++$step) {
+                if (!normalizer_is_normalized($s, self::NFKD)) {
+                    $s = normalizer_normalize($s, self::NFKD);
+                }
+            } elseif (2 === $step) {
+                $s = str_replace(self::TRANSLIT_FROM, self::TRANSLIT_TO, $s);
+            } elseif (3 === $step && '' !== $rule = strtolower(array_shift($rules))) {
+                $step = 2;
+
+                if ('[:nonspacing mark:] remove' === $rule) {
+                    $s = preg_replace('/\p{Mn}++/u', '', $s);
+                } elseif ('de-ascii' === $rule) {
+                    $s = preg_replace("/([AUO])\u{0308}(?=\p{Ll})/u", '$1e', $s);
+                    $s = str_replace(["a\u{0308}", "o\u{0308}", "u\u{0308}", "A\u{0308}", "O\u{0308}", "U\u{0308}"], ['ae', 'oe', 'ue', 'AE', 'OE', 'UE'], $s);
+                } elseif (\function_exists('transliterator_transliterate')) {
+                    if (null === $transliterator = self::$transliterators[$rule] ?? self::$transliterators[$rule] = \Transliterator::create($rule)) {
+                        throw new InvalidArgumentException(sprintf('Unknown transliteration rule "%s".', $rule));
+                    }
+
+                    $s = $transliterator->transliterate($s);
+                }
+            } elseif (!\function_exists('iconv')) {
+                $s = preg_replace('/[^\x00-\x7F]/u', '?', $s);
+            } elseif (\ICONV_IMPL === 'glibc') {
+                $s = iconv('UTF-8', 'ASCII//TRANSLIT', $s);
+            } else {
+                $s = preg_replace_callback('/[^\x00-\x7F]/u', static function ($c) {
+                    $c = iconv('UTF-8', 'ASCII//IGNORE//TRANSLIT', $c[0]);
+
+                    return 1 < \strlen($c) ? ltrim($c, '\'`"^~') : (\strlen($c) ? $c : '?');
+                }, $s);
+            }
+        }
+
+        $str->string .= $s;
+
+        return $str;
+    }
+
+    public function camel(): parent
+    {
+        $str = clone $this;
+        $str->string = str_replace(' ', '', preg_replace_callback('/\b./u', static function ($m) use (&$i) {
```

Comment:
```suggestion
        $str->string = str_replace(' ', '', preg_replace_callback('/\b./u', static function ($m) use (&$i): string {
```
Add another type hint safety guard on anonymous function

Reply:
no need to safeguard internal calls, that'd be just overhead.